### ISSUE TITLE
Feature proposal: Moving EC crypto operation to secp256k1

### ISSUE THREAD

**Issue by NicolasDorier** [MetacoSA/NBitcoin#254]
This mean shipping the binaries of secp256k1 for multiple plateform along with the NBitcoin package.
The hard part is building the secp256k1 on every plateform.

**Comment by dangershony** [comment]
Why the reason of moving to secp256k1?

**Comment by NicolasDorier** [comment]
@dangershony any try of integrating ledger into Breeze ? you will have exactly the same issue.

**Comment by dangershony** [comment]
Porting it will benefit our node and net core we should consider it.

**Comment by NicolasDorier** [comment]
@dangershony I searched briefly, this is a well supported Nuget scenario. The hard thing is to cross compile the binaries.

**Comment by dangershony** [comment]
What you mean is creating a C# wrapper on top of  [secp256k1](https://github.com/bitcoin-core/secp256k1) and publish it as nuget?

**Comment by NicolasDorier** [comment]
yes, and make NBitcoin depends on it.

**Comment by knocte** [comment]
I agree with @dangershony that porting it would be more interesting than providing a binding (but of course I understand if there's no manpower to do it...).

**Comment by lontivero** [comment]
There is a managed wrapper for the secp256k1 curve library: https://github.com/joshlang/Secp256k1.NET
It could be a good starting point. The only thing that doesn't look very well is that the repo doesn't have any activity in two years and the sipa's secp256k1 library is active, in fact it had a schnorr signature implementation months ago that was removed, an other changes.

**Comment by dangershony** [comment]
@knocte we might be able to find funding at stratis I will bring it up, porting will mean we have to maintain the port.

@lontivero that definitely a good starting point I noticed this comment "Sipa's secp256k1 is 100-1000x faster" (then Bouncy Castle)

secp256k1 has a [java wrapper](https://github.com/bitcoin-core/secp256k1/tree/master/src/java) it seems, maybe doing something similar with with C# will be accepted.

**Comment by NicolasDorier** [comment]
I have preference as well for having it ported. That said, it is crazy shit crypto and I feel completely helpless to do that https://github.com/ElementsProject/secp256k1-zkp/tree/secp256k1-zkp .
